# **Iris Classification**

First, we have to import the modules which will be used to perform the task.

In [2]:
import pandas as pd



*   Pandas - Used to analyse the given data.




# **Read the data**

Now, we have to load the Iris data and have to read it.

In [4]:
df=pd.read_csv("/content/Iris (1) (1).csv")

This reads the data and stores it in a variable named "df".

In [5]:
df

Id  SepalLengthCm  ...  PetalWidthCm         Species
0      1            5.1  ...           0.2     Iris-setosa
1      2            4.9  ...           0.2     Iris-setosa
2      3            4.7  ...           0.2     Iris-setosa
3      4            4.6  ...           0.2     Iris-setosa
4      5            5.0  ...           0.2     Iris-setosa
..   ...            ...  ...           ...             ...
145  146            6.7  ...           2.3  Iris-virginica
146  147            6.3  ...           1.9  Iris-virginica
147  148            6.5  ...           2.0  Iris-virginica
148  149            6.2  ...           2.3  Iris-virginica
149  150            5.9  ...           1.8  Iris-virginica

[150 rows x 6 columns]

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id             150 non-null    int64  
 1   SepalLengthCm  150 non-null    float64
 2   SepalWidthCm   150 non-null    float64
 3   PetalLengthCm  150 non-null    float64
 4   PetalWidthCm   150 non-null    float64
 5   Species        150 non-null    object 
dtypes: float64(4), int64(1), object(1)
memory usage: 7.2+ KB


*   The given data contains 150 rows and 6 columns.
*   No data is missing in any row or column.
*   SepalLength, SepalWidth, PetalLength, PetalWidth are in float format.
*   Species are in object format.

In [7]:
df.describe()

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm
count  150.000000     150.000000    150.000000     150.000000    150.000000
mean    75.500000       5.843333      3.054000       3.758667      1.198667
std     43.445368       0.828066      0.433594       1.764420      0.763161
min      1.000000       4.300000      2.000000       1.000000      0.100000
25%     38.250000       5.100000      2.800000       1.600000      0.300000
50%     75.500000       5.800000      3.000000       4.350000      1.300000
75%    112.750000       6.400000      3.300000       5.100000      1.800000
max    150.000000       7.900000      4.400000       6.900000      2.500000

This data shows that every value ranges from (1,10) except the Id(This column will be excluded).

# **Split the Data into Features(x) and Labels(y)**


*   Features are stored in x.
*    In the given data features are SepalLength, SepalWidth, PetalLength, PetalWidth.



In [8]:
x=df.iloc[:,1:-1].values

In [10]:
x

array([[5.1, 3.5, 1.4, 0.2],
       [4.9, 3. , 1.4, 0.2],
       [4.7, 3.2, 1.3, 0.2],
       [4.6, 3.1, 1.5, 0.2],
       [5. , 3.6, 1.4, 0.2],
       [5.4, 3.9, 1.7, 0.4],
       [4.6, 3.4, 1.4, 0.3],
       [5. , 3.4, 1.5, 0.2],
       [4.4, 2.9, 1.4, 0.2],
       [4.9, 3.1, 1.5, 0.1],
       [5.4, 3.7, 1.5, 0.2],
       [4.8, 3.4, 1.6, 0.2],
       [4.8, 3. , 1.4, 0.1],
       [4.3, 3. , 1.1, 0.1],
       [5.8, 4. , 1.2, 0.2],
       [5.7, 4.4, 1.5, 0.4],
       [5.4, 3.9, 1.3, 0.4],
       [5.1, 3.5, 1.4, 0.3],
       [5.7, 3.8, 1.7, 0.3],
       [5.1, 3.8, 1.5, 0.3],
       [5.4, 3.4, 1.7, 0.2],
       [5.1, 3.7, 1.5, 0.4],
       [4.6, 3.6, 1. , 0.2],
       [5.1, 3.3, 1.7, 0.5],
       [4.8, 3.4, 1.9, 0.2],
       [5. , 3. , 1.6, 0.2],
       [5. , 3.4, 1.6, 0.4],
       [5.2, 3.5, 1.5, 0.2],
       [5.2, 3.4, 1.4, 0.2],
       [4.7, 3.2, 1.6, 0.2],
       [4.8, 3.1, 1.6, 0.2],
       [5.4, 3.4, 1.5, 0.4],
       [5.2, 4.1, 1.5, 0.1],
       [5.5, 4.2, 1.4, 0.2],
       [4.9, 3



*   Labels are stored in y
*   In the given data labels are Species




In [11]:
y=df.iloc[:,-1]

In [12]:
y

0         Iris-setosa
1         Iris-setosa
2         Iris-setosa
3         Iris-setosa
4         Iris-setosa
            ...      
145    Iris-virginica
146    Iris-virginica
147    Iris-virginica
148    Iris-virginica
149    Iris-virginica
Name: Species, Length: 150, dtype: object

## **Separating the data into Train and Test**

Importing train_test_split from scikitlearn to perform separating task.

In [14]:
from sklearn.model_selection import train_test_split

In [15]:
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.4,random_state=42)

Now the data has been separated into train and test form.
Let's check for the shape of train variables.

In [18]:
x_train.shape,y_train.shape

((90, 4), (90,))

As we have got the correct values for the size of train variables we may proceed further.

# **Classification using KNN Algorithm**

From scikitlearn we have to import KNN classifier to proceed.

In [19]:
from sklearn.neighbors import KNeighborsClassifier

In [22]:
knc=KNeighborsClassifier(n_neighbors=5,metric="euclidean")

Now we have to train our program with the train variables.

In [23]:
knc.fit(x_train,y_train)

KNeighborsClassifier(algorithm='auto', leaf_size=30, metric='euclidean',
                     metric_params=None, n_jobs=None, n_neighbors=5, p=2,
                     weights='uniform')

As we have trained our program now we'll test this program on the test data

In [36]:
y_pred= knc.predict(x_test)

To find the accuracy and confusion matrix we have to import their respective algorithms from scikitlearn.

In [26]:
from sklearn.metrics import accuracy_score, confusion_matrix

In [27]:
accuracy_score(y_test,y_pred)

0.9833333333333333

In [28]:
confusion_matrix(y_test,y_pred)

array([[23,  0,  0],
       [ 0, 19,  0],
       [ 0,  1, 17]])

We can thus notice that the accuracy for the KNN Classification is 98.3%

# **Classification using DecisionTree**

At first we need to import decision tree from scikitlearn.

In [29]:
from sklearn.tree import DecisionTreeClassifier

In [31]:
dts=DecisionTreeClassifier(random_state=42)

Now, we need to train our program with train variables.

In [32]:
dts.fit(x_train,y_train)

DecisionTreeClassifier(ccp_alpha=0.0, class_weight=None, criterion='gini',
                       max_depth=None, max_features=None, max_leaf_nodes=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=1, min_samples_split=2,
                       min_weight_fraction_leaf=0.0, presort='deprecated',
                       random_state=42, splitter='best')

As we have trained our program now we'll test this program on the test data

In [35]:
y_pred_dts=dts.predict(x_test)

Calculating the accuracy.

In [37]:
accuracy_score(y_test,y_pred_dts)

0.9666666666666667

In [38]:
confusion_matrix(y_test,y_pred_dts)

array([[23,  0,  0],
       [ 0, 18,  1],
       [ 0,  1, 17]])

Thus we have concluded that the accuracy in Decision Tree is 96.6%